src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py

```diff
@@ -65,13 +65,27 @@ def persist(self, cosmos_client: ContainerProxy, blob_container_client: Containe
         from azure.cosmos.exceptions import CosmosResourceExistsError
 
         try:
-            return cosmos_client.create_item(body=self.to_dict())
+            return cosmos_client.create_item(body=self.to_cosmosdb_item())
         except CosmosResourceExistsError:
             return
 
     def to_dict(self) -> Dict[str, Any]:
         return {k: v for k, v in self.__dict__.items() if v}
 
+    def to_cosmosdb_item(self, max_attr_value_length: int = 2048):
+        """
+        Convert the object to a dictionary for persistence to CosmosDB.
+        Truncate attribute values to avoid exceeding CosmosDB's 2MB size limit.
+        """
+        item = self.to_dict()  # use to_dict method to get a dictionary representation of the object
+        attributes = item.get("attributes")
+        if attributes:
+            item["attributes"] = {
+                k: (v if not isinstance(v, str) or len(v) <= max_attr_value_length else v[:max_attr_value_length])
```

Comment:
no need to check `len(v) <= max_attr_value_lengeh`

Reply:
Sure.